src/library/scala/util/Using.scala

```diff
@@ -0,0 +1,287 @@
+package scala.util
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import scala.util.control.{ControlThrowable, NonFatal}
+
+/** A utility for performing automatic resource management. It can be used to perform an
+  * operation using resources, after which it will release the resources, in reverse order
+  * of their creation. The resource opening, operation, and resource releasing are wrapped
+  * in a `Try`.
+  *
```

Comment:
I think the doc should explain what a resource is, how the Resource type class is used, what the user needs to do for non-AutoClosables.

Reply:
Since there are two ways of using `Using`, I think this comment (which is the entry-point) should mention both, say why they exist, what the differences are.

Reply:
When you say two ways, to you mean `Using(foo){ _.bar }` vs `for (f <- Using(foo)) yield f.bar`, or `Using(foo){ _.bar }` vs `Using.resource(foo){ _.bar }`?

Reply:
@lrytz

Reply:
I meant `Using(foo)` vs `Using.resource(foo)`, but explaining the possibilities how to use the monadic version is also welcome.